spec/controllers/target_controller_spec.rb

```diff
@@ -59,6 +59,25 @@
           expect(@parsed_body[0]['edit_url']).to eq(edit_target_url(target_1))
         end
       end
+
+      context 'when a target has a match' do
+        let!(:target_1) do
+          create(:target,
+                 length: 1000, latitude: -34.905446, longitude: -56.206291,
+                 topic: 'Football', user: User.find(1))
+        end
+        let!(:target_compatible) do
+          create(:target,
+                 length: 1000, latitude: -34.906048, longitude: -56.205994,
+                 topic: 'Football')
+        end
+        it 'returns compatible targets' do
+          get :index
+          target_1.matches.each do |m|
```

Comment:
if you just want to make sure the targets.matches are assigned
simply do
`expect(assigns(:mathces)).to match_array(target_1.matches)`